### PR TITLE
Implemented test for ir feature

### PR DESCRIPTION
**Pull request by zaqbez39me**
Test for Impact Ratio (IR) feature in #226

### CODE REVIEW

**Comment by zaqbez39me** [comment]
@yegor256 please, check this test out

**Comment by zaqbez39me** [comment]
@yegor256 fixed the test disabling method

**Comment by yegor256** [comment]
@zaqbez39me I suggest you wait until the build is clean and _then_ ping me

**Comment by yegor256** [comment]
@zaqbez39me yes, indeed, try to run `chmod`, it will help

File: tests/metrics/test-ir.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+exit 0
+
+temp=$1
+stdout=$2
+
+{
+  tmp=$(mktemp -d /tmp/XXXX)
+  cd "${tmp}"
+  mkdir -p "${LOCAL}/${temp}"
+  "${LOCAL}/metrics/ir.sh" ./ "${LOCAL}/${temp}/stdout"
```
Comment (yegor256) [inline tests/metrics/test-ir.sh:38]: @zaqbez39me should fail here

File: tests/metrics/test-ir.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+{
+  tmp=$(mktemp -d /tmp/XXXX)
+  cd "${tmp}"
+  rm -rf ./*
+  rm -rf .git
+  git init --quiet .
+  git config user.email 'foo@example.com'
+  git config user.name 'Foo'
+  "${LOCAL}/metrics/ir.sh" ./ "t0"
```
Comment (yegor256) [inline tests/metrics/test-ir.sh:51]: @zaqbez39me should fail here

File: tests/metrics/test-ir.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  touch "${file1}"
+  git add "${file1}"
+  git config commit.gpgsign false
+  git commit --quiet -m "first file"
+  "${LOCAL}/metrics/ir.sh" ./ "t1"
+  touch "${file2}"
+  git add "${file2}"
+  git commit --quiet -m "second file"
+  "${LOCAL}/metrics/ir.sh" ./ "t2"
+  grep "No files in given repo " "t0" # There are no files in repo
```
Comment (yegor256) [inline tests/metrics/test-ir.sh:64]: @zaqbez39me see https://github.com/yegor256/cam/tree/master/metrics

File: tests/metrics/test-ir.sh (hunk 1 above)
Comment (zaqbez39me) [reply tests/metrics/test-ir.sh:38]: @yegor256 you want me to expect the `ir.sh` failing instead of printing anything here?

**Comment by zaqbez39me** [comment]
@yegor256 please check it out. I changed the expected behaviour after your review.

**Comment by yegor256** [comment]
@zaqbez39me looks good, just one small comment

File: tests/metrics/test-ir.sh [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+set -e
+set -o pipefail
+
+# TODO: #259 ENABLE THIS TESTS RIGHT AFTER IMPLEMENTING ir.sh VIA REMOVING `exit 0`
```
[the remark this answers is not in the document]
Comment (zaqbez39me) [reply tests/metrics/test-ir.sh:28]: I see. Thanks, will follow it

**Comment by yegor256** [comment]
@zaqbez39me check this: https://www.yegor256.com/2024/04/01/ping-me-please.html

**Comment by zaqbez39me** [comment]
> @zaqbez39me check this: https://www.yegor256.com/2024/04/01/ping-me-please.html
Sorry, usually I try to stick with this rule

File: metrics/ir.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+set -e
+set -o pipefail
+
+output=$(realpath "$2")
+
+# TODO: #259 REPLACE WITH ACTUAL METRIC CODE
```
Comment (yegor256) [inline metrics/ir.sh:28]: @zaqbez39me remove this TODO

Comment (zaqbez39me) [reply metrics/ir.sh]: @yegor256 removed it

**Comment by yegor256** [comment]
@zaqbez39me thanks!

**Pull request merged**